File: pattern_searching/naive/python/naive_pattern_searching.py
def pattern_search(main_str, sub_str):
    l_main = len(main_str)
    l_sub = len(sub_str)
    for i in range(l_main-l_sub+1):
        cnt=0
        for j in range(l_sub):
            if (main_str[i + j] != sub_str[j]):
                break
            cnt+=1
        if cnt==l_sub:
            print('sub string matched')
            return True
    print('substr not matched')
    return False   


if __name__ == '__main__':
    pattern_search('ababba','xt')
    pattern_search('ababba','ab')
    pattern_search('ababba','babb')
    pattern_search('ababba','ababba')
    pattern_search('ababba','suyash')

File: pattern_searching/naive/cpp/naive_pattern_searching.cpp
#include<iostream>
using namespace std;

int main(){
    string main_str="abaabbbaaa";
    string sub_str = "suy";
    int l_main= main_str.length();
    int l_sub= sub_str.length();
    bool matched= false;
    for (int i=0;i<=l_main-l_sub+1;i++){
        int cnt=0;
        for(int j=0;j<=l_sub;j++){
            if(main_str[i+j]!=sub_str[j]){
                break;
            }
            cnt++;
        }
        if(cnt==l_sub){
            matched=true;
        }
    }
    if (matched==true){
        cout<<"sub string matched!";
    }
    else{
        cout<<"sub string not matched!";
    }
    return 0;
}